### DISCUSSION TITLE
Hierarchies - use case and samples discussion

### DISCUSSION THREAD

**Discussion by BricoleurSoul** [SkepticMystic/breadcrumbs#121] [Ideas]
I am trying to find a good terminology for a breadcrumb hierarchy to use to capture notes/interdependencies in complex systems eg. an organism (or organisation for that matter).  

For example, the 11 organ systems of a human are certainly not "children of the body" but what terms will fit the "universal" needs of woking with complex systems?  I have started using one as per the attached screenshot  but am not entirely comfortable with it.
<img width="522" alt="Screenshot of Obsidian (2021-08-31, 9-31-25 PM)" src="https://user-images.githubusercontent.com/43626234/131613244-a699e6f4-2f10-4037-854d-09c9ea42a313.png">

However, rather than merely receiving concrete suggestions to this, I would like to see various hierarchies that other users have settled on, and the needs/use case that drove their adoption. I have a feeling that gathering a wide range of examples here might be useful for all of us.

**Comment by SkepticMystic** [comment]
Interesting discussion! I am keen to see what structures others use.
What's interesting for me, is that the code sees the field names in the same way; `parent` is no different to `up`, as long as they're both in the `↑` direction. But the use of different names is in applying them in different cases. So some may have universal names, and others may use multiple, specific structures.
I think @HEmile's structure is quite general. Mine is mostly just `up`, `same`, and `down`. This is quite extensible, but loses meaning in more vague cases of structure.

**Comment by hieu6** [comment]
thank you for posting this, i'm so curious what others are using too! 

my main hierarchy centers around august bradley's ppv self-management system. any note can really be tied in using this hierarchy. having everything tie up to my core values helps me remember _why_ i'm doing some boring thing that i can't immediately recall purpose of. big assist for adhd, poor working memory & lack of executive function.

another unexpected benefit of breadcrumbs is the motivation to add explicit hierarchy metadata is driven by the immediate joy of seeing breadcrumbs on a note **now** as opposed to some nebulous _future_ for dataview tables. motivation to upkeep out the wazoo! 

general

`up`
`next` or `previous`
`down`

---

`compass`
`principles` or `cycles` or `reflect`

---
`pillars`
`personal` or `biological` or `professional`

^^^ i create another hierarchy with each of these as the "up"

example structure below `personal`
`house & home` or `purpose` or `relationships` or `finances` etc.

---

`objective`
`milestone` or `key-result`
`project`


---

for note making, inspired by another obsidianite! 

🍅  or 🧻
📝 
📜

**Reply by BricoleurSoul** [reply]
I'm not familiar with bradley's ppv  but a quick glance at the site it looks like another system with similar intent that I dabbled with many years ago but found I couldn't "establish the habit." 

It might be different now with Obsidian rather than just a notebook to help me along, so thanks for the pointer!

My major use right now though is to synthesise material across the multiple domains covered in my library stacks to try to work on a so far  elusive theory ... I'll post a couple more of the hierarchies I use to help me get my mind around an incredibly diverse "ecology of ideas" that I am trying to get my head around... bottom line I am finding that, just as for you, multiple hierarchies in breadcrumbs seems to be key to any success I am to have.  (that said, like your comment,  I am a big fan of dataview for reporting of status and expected value of the notes I am generating )

**Comment by BricoleurSoul** [comment]
A lot of my current work involves secondary work across multiple domains - I find that similar terms can sometimes have quite different meanings/usage in the different domains. Further, for every concept "affinity" I find in my  research work in different domains, I find multiple "attacks" within each domain as there tends to be many "naysayers" to some approaches by colleagues in the same field.

To try to bring these duelling ideas into a format where I can grasp the nettle I've done some small experiments using two hierarchies - one for grasping notes affirming the way I am heading and one for notes calling that direction into question... I am about ready to dive into the deep end after this small test:
<img width="433" alt="Screenshot of Obsidian (2021-09-02, 9-33-53 PM)" src="https://user-images.githubusercontent.com/43626234/131951039-96f7b337-d69c-4d17-b2e5-79c81fb302b3.png">

**Comment by HEmile** [comment]
This is my hierarchy, focused on academic work. It's a bit more in the semantic links sense.
![image](https://user-images.githubusercontent.com/1289165/131976513-1d26cce1-f7d1-4826-b3ec-eb45fe4a2e24.png)

**Reply by SkepticMystic** [reply]
Interesting! I forget that you don't _need_ any children or siblings; parents are enough to fill in the rest

**Reply by BricoleurSoul** [reply]
nor  are top and bottom  [parents , children ] needed if sibling is set up!

Thanks for this pointer HEmile, I never would have stubbled upon it myself and I think it will be useful where I want to create a note that compares two other notes (perhaps definitions that are different across domains or authors work) so I try to reconcile the two with this construct in the connecting note:

- connects:: [[note b]] and [[note c]]

so my hierarchy might simply consist of 
<img width="672" alt="image" src="https://user-images.githubusercontent.com/43626234/132076645-71f1a306-cb97-4e6e-b4ca-206ba1c48e34.png">

ps. HEmile  - thanks again for Juggl! (I'm flagging some links using "supercharged links" plugin so I'm alerted when showing a note that has a relevant Juggl graph working in conjunction with the Breadcrumb matrix )

**Reply by HEmile** [reply]
Indeed @SkepticMystic, I sometimes use explicit sibling links to make it more explicit (forgot to add those in the picture). 

@BricoleurSoul I don't see how you can do without parents or children, though. siblings is bidirectional while parents/children is unidirectional and creates a hierarchy. 

And thank you :)

**Reply by BricoleurSoul** [reply]
> _@BricoleurSoul I don't see how you can do without parents or children, though. siblings is bidirectional while parents/children is unidirectional and creates a hierarchy._

it is exactly that bidirectional link that I am after!

**Use case**: 
I am building a vault but have imported approximately 4000 text notes from an early wiki app on iOS (Trunknotes circa 2010-11).  These notes are all in CamelCase form for links and to be fully useful then at a minimum those relevant to my current work will need to be converted to .md and have the CameCaseconverted to [[links]]. I don't want to do the conversion in mass (I guess that of the 4000 probably less than 1000 might still be relevant to my current work).  I want to tackle these in some sort of priority order - based on yet unknown paths I am taking in my current work.

Temporary solution a breadcrumb that is bidirectional without explicit forcing into parent|child relationship for those I have fair confidence in their ongoing usefulness.

**Example:**

<img width="1348" alt="Screenshot of Obsidian (2021-09-05, 11-09-48 AM)" src="https://user-images.githubusercontent.com/43626234/132137175-9e7617d3-3468-47e4-9ca1-1d3d7b947f7c.png">

The open note is from original wiki - I have converted to .md an added - connects:: construct to the  link to notes of my "new explorations as captured in a different set of folders the Obsidian Vault.  

**Note:**
@HEmile  I think this might be a way to tackle the problem one of your users has brought forward of "same name" notes in different folders not showing in juggl graph?  A manual hack but it may work (I will be testing that sometime in the next couple days for my own use) - there are certainly many notes in the folder tree I have from the old wiki that will have identical names to my new folders in the vault.
**UPDATE** no it doesn’t work for identically named notes 🤨 In example below only the last referenced  "connects::" is carried forward in Beadcrumb Matrix and juggl graph:
<img width="1066" alt="Screenshot of Obsidian (2021-09-05, 2-40-04 PM)" src="https://user-images.githubusercontent.com/43626234/132142123-19cd95f4-6a40-4736-807a-8a2bd77af914.png">

**Reply by HEmile** [reply]
> it is exactly that bidirectional link that I am after!

Oh, okay! Well in that case, it's great, haha. 

> no it doesn’t work for identically named notes 🤨 In example below only the last referenced "connects::" is carried forward in Beadcrumb Matrix and juggl graph:

Yeah, I've not taking into account the folder/notename syntax, so it won't work for sure.

**Comment by verazorita** [comment]
I currently use breadcrumbs as my main navigation: the hierarchy I designed is rather flat so I made that most yaml field work as a parent. This system requieres mostly no sibling/child definition other than using parent/sibling/child, so I get to use the other fields as verbose as I want them to, which makes a good experience overall.

![imagen](https://user-images.githubusercontent.com/75331117/132236426-2521fc0d-a7b8-4512-a68f-cd8c553ee180.png)
![imagen](https://user-images.githubusercontent.com/75331117/132236397-48256ca3-1b4c-4f41-9207-fa5492122261.png)
![imagen](https://user-images.githubusercontent.com/75331117/132236452-fafa3c65-b8c7-4b30-a98b-9b029ac17e29.png)

**Reply by SkepticMystic** [reply]
Nice! I really like the doubling down on multiple hierarchies. I've mostly tried to squish everything into one hierarchy, but this approach seems very useful!

**Reply by BricoleurSoul** [reply]
I see you are using an 'orphan sibling' ("related") - I presume  in a similar fashion to my use of "connected" (also without defined parent or child entry in the hierarchy).  Couldyou give more details of  where and when you use, and how extensively, rather than relying on implied sibling relationships and how useful it is to you?

**Reply by verazorita** [reply]
`related` is a field inherited from Zotero which I found quite useful. Some items have a close relationship without being worthy of having a collection just for them, and that's when it comes handy, like for example when there are different editions of a same book.

The item shown above is from the entry for Adrian Hon's _"A New history of the future in 100 objects,"_ which was edited by the MIT press, and it is the second edition of _"A history of the future in 100 objects,"_ which was self-published. Something similar happens with Scott Berkun's _"Making things happen:"_ the first edition was called _"The Art of Project Management."_

I checked how I was using it so I could answer and surprisigly, this is the main use case.

**Reply by BricoleurSoul** [reply]
Thanks, that is helpful and extends my thinking of how to approach. I don't think I have run across a similar case in my Vault build YET ...  but most certainly will!

In my current usage I have previous editions of books left with "implied" child relationship and current edition with the child relationship made "explicit" but, in the cases I have so far , the name of the work has not changed.

I use my - connects:: [stand alone sibling] construct to connect topics with the same name that I have in different folders (the result of  smashing two personal wikis together (old circa 2010 and my new Obsidian vault circa 2021). I may end up either merging those notes or adding identifying suffix and moving to the same "new system folder."   If so I may pick up your usage (I Kicked the tires on Zotero but never followed up  - perhaps it is time to do so now ... thanks!)

ps. thanks also for reference to Horn's work - it struck a particular cord, as with my current volunteer work with a Sports Hall of Fame/Museum I am heading production of  a video series, to be followed by a published pamphlet , "The History of Lacrosse in 100 Objects" which is based - of course- on the  principle of the BBC classic!  I took a look on Horn on Amazon and was hooked - my copy is on its way!

**Reply by verazorita** [reply]
Your approach of using `connects` have sparkled the idea of using it for linking _also_ similar concepts, as I suffer from migrating an old databse as well. Thank you!

PS: I like obsidian; the people who uses it read as much as I do! Please send a link to the finished product when available!
